### ISSUE TITLE
Template constraint on runTask does not support `ref`

### ISSUE THREAD

**Issue by Geod24** [vibe-d/vibe-core#194]
Example code:
```D
        string hello;
        runTask(&func, hello);
```

Related error:
```
source/agora/node/main.d(80,16): Error: template vibe.core.core.runTask cannot deduce function from argument types !()(void function(ref string a), string), candidates are:
submodules/vibe-core/source/vibe/core/core.d(318,6):        runTask(ARGS...)(void delegate(ARGS) @safe task, auto ref ARGS args)
submodules/vibe-core/source/vibe/core/core.d(323,6):        runTask(ARGS...)(void delegate(ARGS) @system task, auto ref ARGS args)
submodules/vibe-core/source/vibe/core/core.d(328,6):        runTask(CALLABLE, ARGS...)(CALLABLE task, auto ref ARGS args)
  with CALLABLE = void function(ref string),
       ARGS = (string)
  must satisfy the following constraint:
       is(typeof(CALLABLE.init(ARGS.init)))
```

Thanks to the informative error message, the error is pretty obvious:
One can call `func` with a lvalue, but not with an rvalue, and the test with `ARGS.init` uses rvalues.

**Comment by SingingBush** [comment]
I'm seeing same errors in dmd-2.085.1 on Ubuntu with vibe-core-1.18.0:

```
taggedalgebraic 0.11.22: building configuration "library"...
eventcore 0.9.14: building configuration "epoll"...
stdx-allocator 2.77.5: building configuration "library"...
vibe-core 1.18.0: building configuration "epoll"...
../../../../.dub/packages/vibe-core-1.18.0/vibe-core/source/vibe/core/concurrency.d(1273,12): Error: template `vibe.core.core.runTask` cannot deduce function from argument types `!()(void function(void delegate() op) nothrow @system, void delegate())`, candidates are:
../../../../.dub/packages/vibe-core-1.18.0/vibe-core/source/vibe/core/core.d(317,6):        `vibe.core.core.runTask(ARGS...)(void delegate(ARGS) nothrow @safe task, auto ref ARGS args)`
../../../../.dub/packages/vibe-core-1.18.0/vibe-core/source/vibe/core/core.d(322,6):        `vibe.core.core.runTask(ARGS...)(void delegate(ARGS) nothrow @system task, auto ref ARGS args)`
../../../../.dub/packages/vibe-core-1.18.0/vibe-core/source/vibe/core/core.d(327,6):        `vibe.core.core.runTask(CALLABLE, ARGS...)(CALLABLE task, auto ref ARGS args) if (!is(CALLABLE : void delegate(ARGS)) && isNothrowCallable!(CALLABLE, ARGS))`
../../../../.dub/packages/vibe-core-1.18.0/vibe-core/source/vibe/core/core.d(334,6):        `vibe.core.core.runTask(ARGS...)(void delegate(ARGS) @safe task, auto ref ARGS args)`
../../../../.dub/packages/vibe-core-1.18.0/vibe-core/source/vibe/core/core.d(340,6):        `vibe.core.core.runTask(ARGS...)(void delegate(ARGS) @system task, auto ref ARGS args)`
../../../../.dub/packages/vibe-core-1.18.0/vibe-core/source/vibe/core/concurrency.d(1273,12):        ... (7 more, -v to show) ...
../../../../.dub/packages/vibe-core-1.18.0/vibe-core/source/vibe/core/concurrency.d(1283,18): Error: template `vibe.core.core.runWorkerTask` cannot deduce function from argument types `!()(void function(shared(void delegate()) op) nothrow @system, shared(void delegate()))`, candidates are:
../../../../.dub/packages/vibe-core-1.18.0/vibe-core/source/vibe/core/core.d(509,6):        `vibe.core.core.runWorkerTask(FT, ARGS...)(FT func, auto ref ARGS args) if (isFunctionPointer!FT && isNothrowCallable!(FT, ARGS))`
../../../../.dub/packages/vibe-core-1.18.0/vibe-core/source/vibe/core/core.d(516,6):        `vibe.core.core.runWorkerTask(alias method, T, ARGS...)(shared(T) object, auto ref ARGS args) if (isNothrowMethod!(shared(T), method, ARGS))`
../../../../.dub/packages/vibe-core-1.18.0/vibe-core/source/vibe/core/core.d(523,6):        `vibe.core.core.runWorkerTask(FT, ARGS...)(TaskSettings settings, FT func, auto ref ARGS args) if (isFunctionPointer!FT && isNothrowCallable!(FT, ARGS))`
../../../../.dub/packages/vibe-core-1.18.0/vibe-core/source/vibe/core/core.d(530,6):        `vibe.core.core.runWorkerTask(alias method, T, ARGS...)(TaskSettings settings, shared(T) object, auto ref ARGS args) if (isNothrowMethod!(shared(T), method, ARGS))`
../../../../.dub/packages/vibe-core-1.18.0/vibe-core/source/vibe/core/core.d(538,6):        `vibe.core.core.runWorkerTask(FT, ARGS...)(FT func, auto ref ARGS args) if (isFunctionPointer!FT && isCallable!(FT, ARGS) && !isNothrowCallable!(FT, ARGS))`
```

**Comment by s-ludwig** [comment]
This has been fixed by e5d76ea2648e4d371a178a9b9cdf349f3873b3a3 (#279).
